src/gui/codeeditors/qgscodeeditor.h

```diff
@@ -654,6 +661,88 @@ class GUI_EXPORT QgsCodeEditor : public QsciScintilla
 
 Q_DECLARE_OPERATORS_FOR_FLAGS( QgsCodeEditor::Flags )
 
+
+/**
+ * \ingroup gui
+ * \brief A widget which wraps a QgsCodeEditor in additional functionality.
+ *
+ * This widget wraps an existing QgsCodeEditor object in a widget which provides
+ * additional standard functionality, such as search/replace tools. The caller
+ * must create an unparented QgsCodeEditor object (or a subclass of QgsCodeEditor)
+ * first, and then construct a QgsCodeEditorWidget passing this object to the
+ * constructor.
+ *
+ * \note may not be available in Python bindings, depending on platform support
+ *
+ * \since QGIS 3.38
+ */
+class GUI_EXPORT QgsCodeEditorWidget : public QgsPanelWidget
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsCodeEditorWidget, wrapping the specified \a editor widget.
+     *
+     * Ownership of \a editor will be transferred to this widget.
+     */
+    QgsCodeEditorWidget( QgsCodeEditor *editor SIP_TRANSFER, QWidget *parent SIP_TRANSFERTHIS = nullptr );
+
+    /**
+     * Returns the wrapped code editor.
+     */
+    QgsCodeEditor *editor() { return mEditor; }
+
+  public slots:
+
+    /**
+     * Shows the find bar.
+     *
+     * \see hideFind()
+     * \see setFindVisible()
+     */
+    void showFind();
```

Comment:
How about renaming `Find` to `Search` and adding `Bar` ? I think it's more readable. And do the same to other methods.

```suggestion
    void showSearchBar();
```

Reply:
Good idea, done